### PR TITLE
docs: appset PR generator docs fixes

### PR DESCRIPTION
**Pull request by crenshaw-dev**

### CODE REVIEW

File: docs/operator-manual/applicationset/Generators-Pull-Request.md [hunk 1]
```diff
@@ -49,7 +49,6 @@ spec:
         # Labels is used to filter the PRs that you want to target. (optional)
         labels:
         - preview
-  requeueAfterSeconds: 1800
```
Comment (crenshaw-dev) [inline docs/operator-manual/applicationset/Generators-Pull-Request.md:52]: This field isn't even valid here. Not sure why it was included.

Comment (crenshaw-dev) [reply docs/operator-manual/applicationset/Generators-Pull-Request.md:52]: Nevermind, it's just in the wrong place.

**Review by ashutosh16 — APPROVED** [review]
LGTM

**Comment by mabusaa** [comment]
i think this already fixed by this #10642

**Comment by crenshaw-dev** [comment]
@mabusaa resolved conflicts and left a few small changes.  :-)

**Review by terrytangyuan — COMMENTED** [review]
Consider lower case the file names for appsets, e.g. `Generators-Pull-Request.md` -> `generators-pull-request.md`

**Comment by crenshaw-dev** [comment]
@terrytangyuan fair. I think I'll save that for another cleanup PR. There are a lot of naming inconsistencies in the docs right now.

**Review by terrytangyuan — APPROVED** [review]

**Comment by mubarak-j** [comment]
I think it's worth adding an explanation for `requeueAfterSeconds` and when can it be used in the [template spec page](https://argocd-applicationset.readthedocs.io/en/stable/Template/).

**Review by leoluz — APPROVED** [review]

**Pull request merged**